streams/src/main/java/org/apache/kafka/streams/kstream/KStream.java

```diff
@@ -1209,14 +1210,14 @@ void process(final ProcessorSupplier<? super K, ? super V> processorSupplier,
      * internally generated name, and "-changelog" is a fixed suffix.
      * You can retrieve all generated internal topic names via {@link KafkaStreams#toString()}.
      *
-     * @param otherStream     the {@code KStream} to be joined with this stream
+     * @param otherStream     the {@code KStream}  (side input) to be joined with this (main input) stream
      * @param joiner          a {@link ValueJoiner} that computes the join result for a pair of matching records
      * @param windows         the specification of the {@link JoinWindows}
      * @param keySerde        key serdes for materializing the other stream,
      *                        if not specified the default serdes defined in the configs will be used
-     * @param thisValSerde    value serdes for materializing this stream,
+     * @param thisValSerde    value serdes for materializing one input stream,
```

Comment:
Agree. However, I think side input term (maybe?) can confuse some people, because it is used a bit different in other communities, like Apache Beam. Do you think this can be an issue?